qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -298,4 +340,114 @@ private static void doCodeActionsForUndefinedSectionTag(Template template, Diagn
+		int offset = template.offsetAt(diagnostic.getRange().getStart());
+		Node node = QutePositionUtility.findBestNode(offset, template.findNodeAt(offset));
+
+		Range diagnosticRange = diagnostic.getRange();
+
+		Range rangeValue = new Range(
+				new Position(diagnosticRange.getStart().getLine(), diagnosticRange.getStart().getCharacter()),
+				new Position(diagnosticRange.getEnd().getLine(), diagnosticRange.getEnd().getCharacter()));
+
+		Collection<String> availableValues = new TreeSet<String>(Collator.getInstance());
+
+		collectAvailableValues(node, template, javaCache, availableValues);
+
+		Collection<String> similarValues = new TreeSet<String>(Collator.getInstance());
+
+		for (String value : availableValues) {
+			if (isSimilar(value, varName)) {
+				similarValues.add(value);
+			}
+		}
+		if (!similarValues.isEmpty()) {
+			// Add code actions for each similar value
+			for (String similarValue : similarValues) {
+				CodeAction similarCodeAction = CodeActionFactory.replace("Did you mean '" + similarValue + "'?",
+						rangeValue, similarValue, template.getTextDocument(), diagnostic);
+				codeActions.add(similarCodeAction);
+			}
+		}
+	}
+
+	private static void collectAvailableValues(Node node, Template template, JavaDataModelCache javaCache,
+			Collection<String> availableValues) {
+
+		if (node.getKind() == NodeKind.Expression || node.getKind() == NodeKind.ExpressionParts
+				|| node.getKind() == NodeKind.ExpressionPart) {
+			Expression expression = null;
+			Node nodeExpression = null;
+			if (node.getKind() == NodeKind.Expression) {
+				expression = (Expression) node;
+			} else if (node.getKind() == NodeKind.ExpressionParts) {
+				nodeExpression = node;
+				expression = ((Parts) node).getParent();
+			} else if (node.getKind() == NodeKind.ExpressionPart) {
+				nodeExpression = node;
+				expression = ((Part) node).getParent().getParent();
+			}
+			collectAvailableValuesForExpression(node, template, javaCache, expression, nodeExpression, availableValues);
+		}
+	}
+
+	private static void collectAvailableValuesForExpression(Node node, Template template, JavaDataModelCache javaCache,
+			Expression expression, Node nodeExpression, Collection<String> availableValues) {
+		if (node.getKind() == NodeKind.ExpressionPart) {
+			Part part = (Part) nodeExpression;
+			if (part != null) {
+				List<ValueResolver> namespaceResolvers = javaCache.getNamespaceResolvers(part.getNamespace(),
+						template.getProjectUri());
+				for (ValueResolver resolver : namespaceResolvers) {
+					String label = resolver.getNamespace() + ':' + resolver.getNamed();
+					availableValues.add(label);
+				}
+			}
+		}
+		collectAvailableValuesForObjectPart(node, template, javaCache, expression, nodeExpression, availableValues);
+	}
+
+	private static void collectAvailableValuesForObjectPart(Node node, Template template, JavaDataModelCache javaCache,

```

Comment:
![similar-values-bug2](https://user-images.githubusercontent.com/1417342/179306527-6a028c3b-f562-408b-ac5a-b3d6cd8ecd6a.gif)


This doesn't work for the values of some object. The reason is because you aren't collecting the available values of the Object part. It's currently stored under `nodeExpression` of this method but it isn't used here at all.

I would look at `javaCache.resolveJavaType(..)` as those might do what you need.